### ISSUE TITLE
Support for semantic descriptor options

### ISSUE THREAD

**Issue by fosdev** [alps-io/spec#8]
I think the idea of values (options) for a descriptor is an idea that I think should be in ALPS vs. an extension like https://github.com/alps-io/spec/blob/master/draft-00.txt#L978. One could then have a profile for that information, etc. 

I think ALPS reflects everything one could expect for understanding a profile except for being aware that there is some constrained list of values for a descriptor associated with one.

**Comment by fosrias** [comment]
@mamund So, in the recent google group discussion, it got my wheels turning about this topic again. It seems to me that options are semantics too and could possibly be used reused elsewhere, so they are probably better as descriptors than values in an `ext` element.

Not sure I have gotten all the way there, but the following is grappling with an idea.

```xml
<alps version="1.0">
  <descriptor id="color" type="semantic" rt="#colors"/>
  <descriptor id="blue" type="semantic"/>
  <descriptor id="red" type="semantic"/>
  <descriptor id="colors" type="enum">
    <descriptor href="#blue"/>
    <descriptor href="#green"/>
  </descriptor>
</alps>
```
Here notice introducing a type enum that indicates a list. We could add metadata or other info using `ext` `value` property on the individual color descriptors.

I debated adding an `enum="#colors"` property to a descriptor vs `rt="#colors"` but given `rt` currently is only scoped to transitions, I thought it might make sense to extend that to semantic descriptors as what you MAY get for that value or SHOULD/MAY use for that value.

I think we could add something along these lines as a MAY so we don't force every API to use that list, but then this makes for an easy way to extend the profile at runtime possibly using something like:

```xml
<descriptor href="#color" rt="#my-colors"/>
```

Make any sense?

**Comment by fosrias** [comment]
So, I think this could be solved using an optional `data` property, e.g. `data="enum"`

**Comment by mamund** [comment]
or `<descriptor id="red" type="semantic" enum="true" />

or `<descriptor id="colors" type="semantic" enum="red blue green" />`

or 

```xml
<descriptor id="colors" type="semantic">
  <ext id="colorEnum" type="enum" value="red blue green" />
</descriptor>
```
note in the last solution, `type` property is added to the `ext` element. i think this is an important change we need to do not withstanding this particular change request.

**Comment by fosrias** [comment]
@mamund I think `type` is a good idea and one could get a lot of mileage out of that for data annotations. I think you could also do something like:
```xml
<descriptor id="collection", type="safe", rt="contact">
  <ext id="list" type="array" />
</descriptor>
```
etc.

**Comment by mamund** [comment]
Yep. And can help with reuse in general. I'll write up a change for the spec and we can review. 

Thanks

**Comment by fosrias** [comment]
@mamund I will note that for data annotations, that will get really verbose. And so adding `ext` to every descriptor would be a pain. 

I think or `xml` this would be fairly verbose, but less so for JSON.

```
<descriptor ...>
  <ext href="#list-extension />
  <descriptor id="color" >
     <ext href="#colors-enum"/>
  </descriptor>
<descriptor />
```
Just spit balling, but his might be interesting the ability to ref inline:
<descriptor ext="#list-extension" >
  <descriptor id="color" ext=#colors-enum"/>
<descriptor />

**Comment by mamund** [comment]
Not getting it yet but maybe some larger examples will help me. Feel free to write it up and we can discuss.
